### PR TITLE
Minor improvements to `generate_randoms.py`

### PR DESCRIPTION
**Pull request by bzinberg**
## Summary

- Allow user to specify which datasets to generate
- Allow user to specify where to write the output data
- Change global variables to local where possible

## Tested
Smoke-tested at the command line:
- Ran with no arguments, produced same behavior as before the change
- Specified a subset of the datasets, only those were created
- Specified a different output dir, it was used

### CODE REVIEW

File: pco_cli/generate_randoms.py [hunk 1]
```diff
@@ -1,203 +1,296 @@
...
+@writer
+def write_i64(arr, name, base_dir):
   if arr.dtype != np.int64:
     arr = np.floor(arr).astype(np.int64)
   strs = [str(x) for x in arr]
   full_name = f'i64_{name}'
-  write_generic(strs, arr, full_name)
+  write_generic(strs, arr, full_name, base_dir)
 
-def write_bool(arr, name):
```
Comment (mwlon) [inline pco_cli/generate_randoms.py:45]: I think we can just delete this one now - it's unused

File: pco_cli/generate_randoms.py [hunk 2]
```diff
@@ -1,203 +1,296 @@
 # python 3
 # pip requirement: numpy, pyarrow
```
[the remark this answers is not in the document]
Comment (bzinberg) [reply pco_cli/generate_randoms.py]: Agreed.  Do you want to make a [`requirements.txt`](https://pip.pypa.io/en/stable/user_guide/#requirements-files) or just leave it as a comment here?  (Or is this already a standard format of comment that gets read programmatically?)

File: pco_cli/generate_randoms.py [hunk 3]
```diff
@@ -1,203 +1,296 @@
...
+
+DATA_GENS = {}
+def datagen(*dtypes):
+  def decorate(f):
+    name = f.__name__
+    DATA_GENS[name] = (f, dtypes)
+    return f
+  return decorate
 
 np.random.seed(0)
```
[the remark this answers is not in the document]
Comment (bzinberg) [reply pco_cli/generate_randoms.py]: Whoops, I meant to remove this one too lol.  Thanks for catching.

Re-seeding happens in the main loop, here: https://github.com/mwlon/pcodec/blob/89cb72fc1342aff942e82cc2bb4861f6050d2962/pco_cli/generate_randoms.py#L291-L294

There is one place in this PR where RNG control is not exactly the same as it was before the change: `interl1_scrambl1` calls `interl1_helper` and then uses additional randomness.  To make it behave exactly the same, we would add a `np.random.seed(0)` between these lines: https://github.com/mwlon/pcodec/blob/89cb72fc1342aff942e82cc2bb4861f6050d2962/pco_cli/generate_randoms.py#L227-L228

But I would avoid that unless we want exact backward compatibility (I think we don't care?), as imo a pseudorandom function shouldn't tamper with its own RNG.  If what we want is just that adding a new dataset doesn't change the RNG state seen by existing datasets, then I think the current state of this PR, with a re-seed in the main loop, is good enough, LMKWYT.

File: pco_cli/generate_randoms.py (hunk 1 above)
Comment (bzinberg) [reply pco_cli/generate_randoms.py:45]: Done.

**Comment by mwlon** [comment]
Thanks for your first contribution!

**Pull request merged**